### PR TITLE
LazyLoadingMetadataFactory looks broken

### PR DESCRIPTION
**Pull request by angelk**
| Q               | A
| ------------- | ---
| Branch?       |  3.2
| Bug fix?      | no (tests only)
| New feature?  | no
| BC breaks?    | yes/no
| Deprecations? | no
| Tests pass?   | no
| License       | MIT

I'm expecting to have all parent classes in constraint group. By the [docs] (http://symfony.com/doc/current/validation/groups.html) 
> If you have inheritance (e.g. User extends BaseUser) and you validate with the class name of the subclass (i.e. User), then all constraints in the User and BaseUser will be validated. However, if you validate using the base class (i.e. BaseUser), then only the default constraints in the BaseUser class will be validated.

Testing with LazyLoadingMetadataFactory from v3.2.1 works fine.

### CODE REVIEW

**Comment by xabbuh** [comment]
This looks like it is caused by the changes done in #21053. I am not sure if your use case really makes much sense to be honest.

File: src/Symfony/Component/Validator/Tests/Mapping/Factory/LazyLoadingMetadataFactoryTest.php [hunk 1]
```diff
@@ -168,6 +168,22 @@ public function testMetadataCacheWithRuntimeConstraint()
...
         $metadata = $factory->getMetadataFor(self::CLASS_NAME);
     }
+
+    public function testGroupsFromParent()
+    {
+        $reader = new \Symfony\Component\Validator\Mapping\Loader\StaticMethodLoader();
+        $factory = new LazyLoadingMetadataFactory($reader);
+        $metadata = $factory->getMetadataFor('Symfony\Component\Validator\Tests\Fixtures\EntityStaticCarTurbo');
+        $classMetaData = $metadata->getPropertyMetadata('wheels');
+        $constraints = $classMetaData[0]->getConstraints();
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Mapping/Factory/LazyLoadingMetadataFactoryTest.php]: The culprit here is that you are making the assumption that `getPropertyMetadata()` of the `ClassMetadata` class will online return one `PropertyMetadata` instance (or that the one you need is the first entry in the returned collection). It's true that this was the case before #20793, but that IMO is can implementation detail you shouldn't rely on (the `RecursiveContextualValidator`, for example, iterates over the return property metadata). I'll see if we can nonetheless update the code to only return one instance, but as said that's not something I would rely on.

**Comment by angelk** [comment]
Hello,
I just updated the tests to iterate over constraints and groups. 
Two groups are still missing in metadata - https://travis-ci.org/symfony/symfony/jobs/200139678#L3805

**Comment by fabpot** [comment]
fabbot failures must be fixed.

**Comment by xabbuh** [comment]
Thank you for the failing test cases @angelk. I have reused them to revert the changes from #21053 in #21592. Closing here in favour of #21592.

**Pull request closed without merging**